kuwala/core/neo4j/importer/src/Neo4jConnection.py

```diff
@@ -46,39 +44,26 @@ def query_graph(q, parameters=None, db=None):
     return response, retry
 
 
-# Method used inside Pyspark
-# Establish new database connection for each invocation due to multi-threaded environment
-def batch_insert_data(partition, query):
-    def send_query(rows):
-        sleep_time = 1
-
-        while True:
-            response, retry = query_graph(query, parameters={'rows': rows})
-
-            if not retry:
-                break
-            elif sleep_time < 60:
-                sleep(sleep_time)
-                sleep_time *= 2
-            else:
-                return response
-
-    connect_to_graph(uri="bolt://localhost:7687",
-                     user="neo4j",
-                     password="password")
-
-    batch = list()
-    batch_size = 10000
-
-    for row in partition:
-        batch.append(row.asDict())
-
-        if len(batch) == batch_size:
-            send_query(batch)
-
-            batch = list()
-
-    if len(batch) > 0:
-        send_query(batch)
-
-    close_connection()
+def spark_send_query(df, query):
```

Comment:
I would change the name of the function here to describe exactly wat it does... write_df_to_neo4j_with_override(..)